vme/zone/competition.zon

```diff
@@ -128,6 +130,31 @@ code
 }
 dilend
 
+dilbegin auto_xp_death (); // Automatically gets xp and deaths data from characters every ten minutes for tracking
+external 
+	upd_xp_char(u:unitptr);
+	upd_dead_ch(u:unitptr);
+	string rlc_format (sl:stringlist,s:string);
+var
+	rm:unitptr;
+	u:unitptr;
+	i,ln:integer;
+code
+{
+	:start:
+	heartbeat:=PULSE_SEC*600;
+	pause;
+	u := ghead();
+	while (u.type==UNIT_ST_PC)
+	{
+		upd_xp_char(u);
+		upd_dead_ch(u);
+		u:=u.gnext;
```

Comment:
You should break out of the loop as soon as the type is != UNIT_ST_PC then you'll save ~8000 checks.  That works because PCs are always at the head of the list.